### ISSUE TITLE
Not sure how to draw lines

### ISSUE THREAD

**Issue by PenelopeFudd** [mozilla/seasponge#70]
I can create threats, and drag and drop stencils.  The thing I can't figure out is how to connect the stencils with lines.  If a person has to read the documentation to figure that out, it may be that the UI needs some more refinement.

**Comment by PenelopeFudd** [comment]
Aha.  The animated gif made it clear; the solid circles on the edges of stencils are targets, the empty circles are sources.  Could you make the empty circles into arrow heads?

**Comment by Glavin001** [comment]
Have you taken a look at https://github.com/mozilla/seasponge/wiki/Usage ?

![](https://cloud.githubusercontent.com/assets/5893112/6827282/55cf9060-d2e5-11e4-802f-7663719ee873.gif)

/cc @Frozenfire92

**Comment by PenelopeFudd** [comment]
Yes, the animated gif on the Usage page made it clear, thanks.  I'm just saying that the best programs I've seen didn't need (external) documentation for their basic functions.  Also, UI design is hard.  Also^2, seeing what difficulties absolutely new users encounter is the key to getting the UI perfect.

Thanks for a great program!

**Comment by Glavin001** [comment]
Absolutely agree. The goal is that the usage documentation is not even necessary, and we thought that had been achieved. However, I can understand how it may be confusing and we do want to make those endpoints more obvious that they are draggable.

Another idea is that we get rid of the endpoints entirely and make the stencil element itself like an endpoint that you can just create and connect to. See http://www.jsplumb.org/demo/sourcesAndTargets/dom.html for example.

/cc @Frozenfire92  what are your thoughts on getting rid of the endpoints in favour of something like http://www.jsplumb.org/demo/sourcesAndTargets/dom.html ?

**Comment by Glavin001** [comment]
I think removing the anchor/endpoints and making the stencil elements themselves have the drag-n-drop flows, would be easiest to understand and intuitive.

![demo](
![image](https://cloud.githubusercontent.com/assets/1885333/6966940/a0f742f6-d932-11e4-81a7-5707937b9fa2.png))


Thoughts?

**Comment by Frozenfire92** [comment]
> what are your thoughts on getting rid of the endpoints in favour of something like http://www.jsplumb.org/demo/sourcesAndTargets/dom.html ?

I think this is an interesting idea, however I'm not sure on this solving the underlying problem of knowing which way the data flows. For example that one you can only go blue to green, but the way our app works you can go any element to any element. What signifies that direction?

**Comment by Glavin001** [comment]
jsPlumb supports having arrow overlays to signify direction.

- Docs: https://jsplumbtoolkit.com/doc/overlays.html
- Demo: http://www.freedevelopertutorials.com/jsplumb-tutorial/examples/jsplumb-connection-arrow-example/
